Fix coding error in procedure TLinkerBSD.SetDefaultInfo in new patch.
Stop adding nested "(((s<>'c') and (s<>'iconv'))" too (lang/fpc/Makefile:post-patch:ICONV_LIB)
Use systems_dragonfly for consistency.

--- files/patch-compiler_systems_t__bsd.pas.orig	2019-09-04 12:05:52 UTC
+++ files/patch-compiler_systems_t__bsd.pas
@@ -44,28 +44,19 @@
        DynamicLinker:='/usr/libexec/ld.so'
       else if target_info.system in systems_netbsd then
        DynamicLinker:='/usr/libexec/ld.elf_so'
-+     else if target_info.system=system_x86_64_dragonfly then
++     else if target_info.system in systems_dragonfly then
 +      DynamicLinker:='/libexec/ld-elf.so.2'
       else
         DynamicLinker:='';
     end;
-@@ -555,7 +557,7 @@
-      While not SharedLibFiles.Empty do
-       begin
-         S:=SharedLibFiles.GetFirst;
--        if (s<>'c') or reorder then
-+        if (((((((s<>'c') and (s<>'iconv')) and (s<>'iconv')) and (s<>'iconv')) and (s<>'iconv')) and (s<>'iconv')) and (s<>'iconv')) or reorder then
-          begin
-            i:=Pos(target_info.sharedlibext,S);
-            if i>0 then
 @@ -718,7 +720,12 @@
    Replace(cmdstr,'$DYNLINK',DynLinkStr);
    if (target_info.system in systems_darwin) then
      Replace(cmdstr,'$PRTOBJ',GetDarwinPrtobjName(false));
 -  BinStr:=FindUtil(utilsprefix+BinStr);
-+  if target_info.system=system_x86_64_dragonfly then
-+     BinStr:='/usr/libexec/binutils225/elf/ld.bfd';
-+  if target_info.system in systems_freebsd then
++  if target_info.system in systems_dragonfly then
++     BinStr:='/usr/bin/ld.bfd'
++  else if target_info.system in systems_freebsd then
 +     BinStr:='%%LDPATH%%'
 +  else
 +     BinStr:=FindUtil(utilsprefix+BinStr);
